backend/tests/api/test_users.py

```diff
@@ -0,0 +1,132 @@
+from app.auth.config import API_SECRET_KEY
+from app.users.routers import router
+from fastapi.testclient import TestClient
+
+client = TestClient(router)
+
+
+def test_create_user(client: TestClient) -> None:
+    headers = {
+        "accept": "application/json",
+        "Authorization": f"Bearer {API_SECRET_KEY}",
+    }
+
+    user_data = {
+        "email": "testuser@example.com",
+        "password": "securepassword",
+        "role": "user",
+    }
+
+    response = client.post("/users/create", json=user_data, headers=headers)
+    assert response.status_code == 200
+    json_response = response.json()
+    assert json_response["email"] == user_data["email"]
+    assert json_response["action_taken"] == "created"
+
+
+def test_get_user(client: TestClient) -> None:
+    headers = {
+        "accept": "application/json",
+        "Authorization": f"Bearer {API_SECRET_KEY}",
+    }
+
+    user_data = {
+        "email": "testuser2@example.com",
+        "password": "securepassword",
+        "role": "user",
+    }
+    create_response = client.post("/users/create", json=user_data, headers=headers)
```

Comment:
Should have a non-admin try to create this and fail

Reply:
I think - for now - yes?